mnelab/model.py

```diff
@@ -131,22 +131,39 @@ def find_events(self, stim_channel, consecutive=True, initial_event=True,
     @data_changed
     def events_from_annotations(self):
         """Convert annotations to events."""
-        events, _ = mne.events_from_annotations(self.current["data"])
+        events, mappings = mne.events_from_annotations(self.current["data"])
+        # Mappings for annots are swapped {str: int} should be {int: str}
+        mappings = dict((v, k) for k, v in mappings.items())
         if events.shape[0] > 0:
             self.current["events"] = events
-            self.history.append("events, _ = "
+            self.current["event_mappings"] = mappings
+            self.history.append("events, event_mappings = "
                                 "mne.events_from_annotations(data)")
 
     @data_changed
     def annotations_from_events(self):
         """Convert events to annotations."""
-        annots = mne.annotations_from_events(
-            self.current["events"],
-            self.current["data"].info["sfreq"]
-        )
+        if "event_mappings" in self.current:
+            annots = mne.annotations_from_events(
+                self.current["events"],
+                self.current["data"].info["sfreq"],
+                event_desc=self.current["event_mappings"]
+            )
+        else:
+            annots = mne.annotations_from_events(
+                self.current["events"],
+                self.current["data"].info["sfreq"],
+            )
```

Comment:
This if/else block could be shortened because the only difference is the `event_desc` argument, which you could set to `None` in case it is not available and then you can use one function call. Same for the block further down in line 159.

Reply:
Thanks for the suggestion! I hadn't thought about it. I followed the example of what you did on the `plot_data` function.